src/App.jsx

```diff
@@ -14,33 +16,142 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABET = 'alphabet';
+const SORT_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function sortByParameter(goods, { param = '', reverse }) {
+  let goodsArr = goods;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (param) {
+    case SORT_ALPHABET: {
+      goodsArr = goods.sort((good1, good2) => {
+        return good1.localeCompare(good2);
+      });
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      break;
+    }
+
+    case SORT_LENGTH: {
+      goodsArr = goods.sort((good1, good2) => {
+        return good1.length - good2.length;
+      });
+
+      break;
+    }
+
+    default: {
+      break;
+    }
+  }
+
+  if (reverse) {
+    const temp = [];
+
+    for (let i = goodsArr.length - 1; i >= 0; i -= 1) {
+      temp.push(goodsArr[i]);
+    }
+
+    return temp;
+  }
+
+  return goodsArr;
+}
+
+export const App = () => {
+  const [sort, setSort] = useState('');
```

Comment:
it's better to create 2 states `sortBy` and `isReversed` (boolean)

then work with it.